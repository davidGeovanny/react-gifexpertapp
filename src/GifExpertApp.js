import React, { Fragment, useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';
import PropTypes from 'prop-types';

const GifExpertApp = ({ defaultCategories }) => {
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = e => {
    //     const newCategory = 'Rising of Shield Hero';

    //     setCategories( [...categories, newCategory] );
    //     // setCategories( category => {
    //     //     return [...category, newCategory];
    //     // });
    // };
    
    return (
        <Fragment>
            <h2>GifExpert</h2>
            <AddCategory 
                setCategories={ setCategories } 
            />
            <hr />

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            category={ category }
                            key={ category }
                        />
                    ))
                }
            </ol>
        </Fragment>
    );
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array
};

GifExpertApp.defaultProps = {
    defaultCategories: []
};
 
export default GifExpertApp;