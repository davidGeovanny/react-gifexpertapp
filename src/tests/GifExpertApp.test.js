import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test with <GifExpertApp />', () => {
    test('should match the <GifExpertApp /> with the snapshot correctly', () => {
        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('should to show a category list', () => {
         const categories = ['Boku No Hero', 'Dragon Ball'];

         const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

         expect( wrapper ).toMatchSnapshot();
         expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
    
     
});
