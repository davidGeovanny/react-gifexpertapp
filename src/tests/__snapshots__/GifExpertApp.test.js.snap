// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test with <GifExpertApp /> should match the <GifExpertApp /> with the snapshot correctly 1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test with <GifExpertApp /> should to show a category list 1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Boku No Hero"
      key="Boku No Hero"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
